src/SWP/Bundle/CoreBundle/Controller/FbPageController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Superdesk Web Publisher Core Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\CoreBundle\Controller;
+
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use SWP\Bundle\CoreBundle\Form\Type\FacebookPageType;
+use SWP\Bundle\CoreBundle\Model\FacebookPage;
+use SWP\Bundle\FacebookInstantArticlesBundle\Model\PageInterface;
+use SWP\Component\Common\Criteria\Criteria;
+use SWP\Component\Common\Pagination\PaginationData;
+use SWP\Component\Common\Response\ResourcesListResponse;
+use SWP\Component\Common\Response\ResponseContext;
+use SWP\Component\Common\Response\SingleResourceResponse;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\ConflictHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class FbPageController extends Controller
+{
+    /**
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists Facebook Pages",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         500="Unexpected error."
+     *     }
+     * )
+     * @Route("/api/{version}/facebook/pages/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_list_facebook_pages")
+     * @Method("GET")
+     */
+    public function listAction(Request $request)
+    {
+        $repository = $this->get('swp.repository.facebook_page');
+
+        $items = $repository->getPaginatedByCriteria(
+            new Criteria(),
+            ['id' => 'asc'],
+            new PaginationData($request)
+        );
+
+        return new ResourcesListResponse($items);
+    }
+
+    /**
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Create Facebook page",
+     *     statusCodes={
+     *         201="Returned on success.",
+     *         400="Returned when not valid data."
+     *     },
+     *     input="SWP\Bundle\CoreBundle\Form\Type\FacebookPageType"
+     * )
+     * @Route("/api/{version}/facebook/pages/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_create_facebook_pages")
+     * @Method("POST")
+     */
+    public function createAction(Request $request)
+    {
+        /* @var FacebookPage $feed */
+        $page = $this->get('swp.factory.facebook_page')->create();
```

Comment:
`$this->get('swp.factory.facebook_page')` - you can assign it to variable as it used more than once in the same method